File: python/dialogs/message_box_yes_no.py
"""Standard Yes/No message box."""

try:
    from dialogs.message_box import MessageBoxBehavior
    from dialogs.message_box_yes_no_ui import MessageBoxYesNoUI
except ImportError:
    from message_box import MessageBoxBehavior
    from message_box_yes_no_ui import MessageBoxYesNoUI


class MessageBoxYesNo(MessageBoxBehavior, MessageBoxYesNoUI):
    def __init__(self, header="Message", message="", on_close=None):
        super().__init__()
        self._configure_message_box(
            header=header,
            message=message,
            on_close=on_close,
            result_by_button={
                "yes_btn": "yes",
                "no_btn": "no",
            },
            default_result="no",
        )

File: java/src/main/java/uimd/MessageBoxYesNo.java
package uimd;

import java.util.Map;
import java.util.function.Consumer;

public final class MessageBoxYesNo extends MessageBoxYesNoUI
{
    private final MessageBoxBehavior behavior;

    public MessageBoxYesNo()
    {
        this("Message", "", null);
    }

    public MessageBoxYesNo(String header, String message)
    {
        this(header, message, null);
    }

    public MessageBoxYesNo(String header, String message, Consumer<String> onClose)
    {
        behavior = new MessageBoxBehavior(
            dialog_header,
            this.message,
            header,
            message,
            onClose,
            Map.of("yes_btn", "yes", "no_btn", "no"),
            "no");
    }

    public String result()
    {
        return behavior.result();
    }

    public boolean closed()
    {
        return behavior.closed();
    }

    public int run(String[] arguments)
    {
        return GeneratedWindowRuntime.runGeneratedWindow(this, runtimeOptions(), arguments);
    }

    @Override
    public GeneratedWindowRuntimeOptions runtimeOptions()
    {
        return behavior.configure(super.runtimeOptions());
    }

    @Override
    protected void onYesBtnClick()
    {
        behavior.handleButton("yes_btn");
    }

    @Override
    protected void onNoBtnClick()
    {
        behavior.handleButton("no_btn");
    }
}
